File: WebAdmin/src/CommuniqueController.js
class CommuniqueController {
    api;
    constructor(){
        this.api = new clsApiConnection();
    }

    async getByLocal(local)
    {
        return await
         this.api.httpGet(`Communique?local=${local}`);
    }

    async add(localId, type, description, start, end)
    {
        return await
         this.api.httpPost(
             'Communique/',
             {
                localId: localId,
                type: type,
                description: description,
                startDate: start,
                endDate: end
             }
         );
    }

    async bindSubCategories(communiqueId, subCategories = [,])
    {
        return await
         this.api.httpPost(
             'Communique/BindSubCategories/',
             {
                communiqueId: communiqueId,
                communiqueSubCategory: [
                    {
                        subCategoryId: 0,
                        category: {
                            categoryId: 2
                        }
                    },
                ]
             }
         );
    }

    async disbindSubCategories(communiqueId, subCategories = [,])
    {
        return await
         this.api.httpPost(
             'Communique/DisbindSubCategories/',
             {
                communiqueId: communiqueId,
                communiqueSubCategory: [
                    {
                        subCategoryId: 0,
                        category: {
                            categoryId: 2
                        }
                    },
                ]
             }
         );
    }

    async update(id, localId, type, description, start, end)
    {
        return await
         this.api.httpPatch(
             'Communique/',
             {
                communiqueId: id,
                localId: localId,
                type: type,
                description: description,
                startDate: start,
                endDate: end
             }
         );
    }

    async delete(id, type)
    {
        return await this.api.httpDelete(
            `Communique?id=${id}&type=${type}`
        );
    }
}

File: MobileApp/lib/Controller/Communique.controller.dart
import 'package:ReachUp/Model/Communique.model.dart';
import 'package:ReachUp/Model/CommuniqueSubCategory.model.dart';
import 'package:ReachUp/Repositories/Communique.repository.dart';

class CommuniqueController {
  CommuniqueRepository communiqueRepository;
  Communique communique;

  CommuniqueController() {
    communiqueRepository = new CommuniqueRepository();
  }

  Future<List<Communique>> getByLocal(int localId, bool general) async {
    return communiqueRepository.byLocal(localId, general);
  }

  Future<bool> add(Communique communique) async {
    return communiqueRepository.add(communique);
  }

  Future<bool> bindSubCategories(List<CommuniqueSubCategory> communiqueSubCategory) async {
    return communiqueRepository.bindSubCategories(communiqueSubCategory);
  }



  // Future<bool> disbindSubCategories(dynamic communique) async {
  //   return communiqueRepository.disbindSubCategories(communique);
  // }

  Future<bool> update(dynamic communique) async {
    return communiqueRepository.update(communique);
  }

  Future<bool> delete(int id, int type) async {
    return communiqueRepository.delete(id, type);
  }
}
